recovery/recovery.py

```diff
@@ -71,73 +70,104 @@ def __init__(self):
+                    'Set name of the file (without file format) to save data to/load data from. Default name is empty. '
+                    'If name is empty: load operation will use the last available json backup file, '
                     'save operation will use date and time in UTC.'
                 ),
                 'type': str,
                 'default': ''
             },
+            {
+                'flag': '--rdb_file',
+                'help': (
+                    'Set name of the file to save data from redis database rdb file to. '
+                    'Default name is current UTC datetime.'
+                ),
+                'type': str,
+                'default': datetime.utcnow().strftime(backup_date_format)
+            },
         ]
         super().__init__(help, args, None if __name__ == '__main__' else [], mutually_exclusive_args)
 
-        self.log_directory = config.get('Directory-Section', 'logs')
-        self.temp_dir = config.get('Directory-Section', 'temp')
-        self.cache_directory = config.get('Directory-Section', 'cache')
-        self.redis_host = config.get('DB-Section', 'redis-host')
-        self.redis_port = config.get('DB-Section', 'redis-port')
-        self.var_yang = config.get('Directory-Section', 'var')
-
 
-def feed_confd_modules(modules: list, confd_service: ConfdService) -> bool:
-    return confd_service.patch_modules(modules)
-
-
-def main(script_conf: BaseScriptConfig = ScriptConfig()):
-    start_time = int(time.time())
-    args = script_conf.args
-    cache_directory = script_conf.cache_directory
-    log_directory = script_conf.log_directory
-    temp_dir = script_conf.temp_dir
-    var_yang = script_conf.var_yang
-    confd_service = ConfdService()
-
-    confd_backups = os.path.join(cache_directory, 'confd')
-    redis_backups = os.path.join(cache_directory, 'redis')
-    redis_json_backup = os.path.join(cache_directory, 'redis-json')
-
-    logger = log.get_logger('recovery', os.path.join(log_directory, 'yang.log'))
-    process_type = 'save' if args.save else 'load'
-    logger.info(f'Starting {process_type} process of Redis database')
-    job_log_filename = 'recovery - save' if args.save else current_file_basename
-    job_log(start_time, temp_dir, status=JobLogStatuses.IN_PROGRESS, filename=job_log_filename)
-    job_log_messages = []
-    if args.save:
-        args.file = args.file or datetime.datetime.utcnow().strftime(backup_date_format)
+class Recovery:
+    def __init__(
+            self,
+            args: Namespace,
+            config: ConfigParser = create_config(),
+            redis_connection: RedisConnection = RedisConnection()
+    ):
+        self.start_time = None
+        self.job_log_messages = []
+        self.job_log_filename = current_file_basename
+        self.yang_catalog_module_name = 'yang-catalog@2018-04-03/ietf'
+        self.process_type = ''
+
+        self.args = args
+        self.config = config
```

Comment:
`config` is not used anywhere outside this `__init__` method so it don't need to be in `self`